### ISSUE TITLE
AdmPushReceiver crash

### ISSUE THREAD

**Issue by gvoltr** [urbanairship/android-library#18]
Hello, I have some NPE crash reports in the fabric caused by `com.urbanairship.push.AdmPushReceiver.onReceive`  
There are 2 crash types one on line 39 and another on line 53 both are NullPointerException.
I'm using latest library version `8.6.+` and play services `11.0.2`
Crash occurs on devices with android version 4.1.1 - 4.3.1
Can somebody please explain, what can cause this issues?

**Comment by ralepinski** [comment]
Could you share the crash report? Is your application configured for ADM or running on an ADM device?

**Comment by rlepinski** [comment]
Could you share the crash report? Is your application configured for ADM or running on an ADM device?

**Comment by gvoltr** [comment]
First type of crash -
```
Fatal Exception: java.lang.RuntimeException: Unable to start receiver com.urbanairship.push.AdmPushReceiver: java.lang.NullPointerException
       at android.app.ActivityThread.handleReceiver(ActivityThread.java:2431)
       at android.app.ActivityThread.access$1500(ActivityThread.java:141)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1332)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:137)
       at android.app.ActivityThread.main(ActivityThread.java:5103)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:525)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:737)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java)
       at dalvik.system.NativeStart.main(NativeStart.java)
Caused by java.lang.NullPointerException
       at com.urbanairship.push.AdmPushReceiver.onReceive(AdmPushReceiver.java:53)
       at android.app.ActivityThread.handleReceiver(ActivityThread.java:2424)
       at android.app.ActivityThread.access$1500(ActivityThread.java:141)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1332)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:137)
       at android.app.ActivityThread.main(ActivityThread.java:5103)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:525)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:737)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java)
       at dalvik.system.NativeStart.main(NativeStart.java)
```

Second type of crash - 
```
Fatal Exception: java.lang.RuntimeException: Unable to start receiver com.urbanairship.push.AdmPushReceiver: java.lang.NullPointerException
       at android.app.ActivityThread.handleReceiver(ActivityThread.java:2431)
       at android.app.ActivityThread.access$1500(ActivityThread.java:141)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1332)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:137)
       at android.app.ActivityThread.main(ActivityThread.java:5103)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:525)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:737)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java)
       at dalvik.system.NativeStart.main(NativeStart.java)
Caused by java.lang.NullPointerException
       at com.urbanairship.push.AdmPushReceiver.onReceive(AdmPushReceiver.java:39)
       at android.app.ActivityThread.handleReceiver(ActivityThread.java:2424)
       at android.app.ActivityThread.access$1500(ActivityThread.java:141)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1332)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:137)
       at android.app.ActivityThread.main(ActivityThread.java:5103)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:525)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:737)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java)
       at dalvik.system.NativeStart.main(NativeStart.java)
```
In my project only GCM is configured. Devices are just ordinary phones like nexus 5, for exmaple.

**Comment by rlepinski** [comment]
How many crashes?

The only way to message that receiver is if the app claims the adm permission. I am pretty sure this is caused by a malicious app trying to message your application. We do permission checks within the SDK, so even if it didn't crash we would drop the intent on the floor if we are not an Amazon device, adm is not available, or we are not registered for adm. 

I'll fix the npe exception today in 8.8.

**Comment by gvoltr** [comment]
About 50 crashes in last 30 days. Thank you.

**Comment by rlepinski** [comment]
Fixed in 8.8
